config/validation.go

```diff
@@ -0,0 +1,66 @@
+package config
+
+import (
+	"fmt"
+	"github.com/contiv/errored"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	gojson "github.com/xeipuuv/gojsonschema"
+)
+
+// Combines array of errors into a single error
+func combineErrors(resultErrors []gojson.ResultError) error {
+	var errors []string
+	for _, err := range resultErrors {
+		errors = append(errors, fmt.Sprintf("%s\n", err))
+	}
+	return errored.New(strings.Join(errors, "\n"))
+}
+
+// ValidateRuntime validates the given runtime against its defined schema
+func ValidateRuntime(cfg *RuntimeOptions) error {
+	schema := gojson.NewStringLoader(RuntimeSchema)
+	doc := gojson.NewGoLoader(cfg)
+
+	if result, _ := gojson.Validate(schema, doc); !result.Valid() {
+		return combineErrors(result.Errors())
+	}
+	log.Infof("Runtime config validation passed")
+	return nil
+}
+
+// ValidatePolicy validates the given policy against its defined schema
+func ValidatePolicy(cfg *Policy) error {
```

Comment:
Combine the methods?

On Tue, Jul 26, 2016 at 9:57 PM, Yuva Shankar <notifications@github.com>
wrote:

> In config/validation.go
> <https://github.com/contiv/volplugin/pull/366#discussion_r72380800>:
>
> > +}
> > +
> > +// ValidateRuntime validates the given runtime against its defined schema
> > +func ValidateRuntime(cfg *RuntimeOptions) error {
> > +	schema := gojson.NewStringLoader(RuntimeSchema)
> > +	doc := gojson.NewGoLoader(cfg)
> > +
> > +	if result, _ := gojson.Validate(schema, doc); !result.Valid() {
> > +		return combineErrors(result.Errors())
> > +	}
> > +	log.Infof("Runtime config validation passed")
> > +	return nil
> > +}
> > +
> > +// ValidatePolicy validates the given policy against its defined schema
> > +func ValidatePolicy(cfg *Policy) error {
>
> there is already a validate in config/policy.go(func (*Policy) Validate())
> . If I define it here, it will be a duplicate.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/contiv/volplugin/pull/366/files/55dc950ae59a1a55ef6b6ba22279d398d6da6f91..de036c928b2bc124e2340acbf576bd0cb54d8d2e#r72380800>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABJ63yg5AF14DCtyI479F7KjlO40CNTks5qZuVPgaJpZM4JVYRC>
> .
>